File: java/src/main/java/helpers/TemperatureConstantGetter.java
package helpers;


import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class TemperatureConstantGetter {

    private final HttpClient httpClient = HttpClient.newHttpClient();

    public int getHotnessConstant() {
        // We call an external service to get the constant, so first, build the request
        var request = HttpRequest.newBuilder(
                URI.create("https://api.nasa.gov/magicConstantValue1"))
                .header("accept", "application/json")
                .build();

        // Then send the request
        return makeRequest(request);
    }

    public int getHumidityConstant() {
        // We call an external service to get the constant, so first, build the request
        var request = HttpRequest.newBuilder(
                URI.create("https://api.nasa.gov/magicConstantValue2"))
                .header("accept", "application/json")
                .build();

        // Then send the request
        return makeRequest(request);
    }

    private int makeRequest(HttpRequest request) {
        HttpResponse<String> response;
        try {
            response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
            return 0;
        }
        return Integer.getInteger(response.body());
    }
}

File: typescript/src/unitTestingInterviewQuestion/helpers/TemperatureConstantGetter.ts
export class TemperatureConstantGetter {
    async getHotnessConstant(): Promise<number> {
        // We call an external service to get the constant, so first, build the request
        var request = new Request('https://api.nasa.gov/magicConstantValue1', {
            method: 'GET',
            headers: {
                accept: 'application/json'
            }
        });
        // Then send the request
        return await this.makeRequest(request);
    }

    async getHumidityConstant(): Promise<number> {
        // We call an external service to get the constant, so first, build the request
        var request = new Request('https://api.nasa.gov/magicConstantValue2', {
            method: 'GET',
            headers: {
                accept: 'application/json'
            }
        });
        // Then send the request
        return await this.makeRequest(request);
    }

    private async makeRequest(request: Request): Promise<number> {
        let response: Response;
        try {
            response = await fetch(request);
        } catch (error) {
            console.error(error);
            return 0;
        }
        return parseInt(await response.text());
    }
}
